### PR TITLE
Batch calls to get_bundle_uuids in worksheet parsing

### PR DESCRIPTION
**Pull request by kashizui**
@percyliang @pbhatnagar3

### CODE REVIEW

**Comment by kashizui** [comment]
whoops need to implement comments

File: codalab/lib/worksheet_util.py [hunk 1]
```diff
@@ -113,12 +112,13 @@ def get_worksheet_info_edit_command(raw_command_map):
...
-    bundle_uuid = bundle_info['uuid'] if bundle_info else None
-    subworksheet_uuid = subworksheet_info['uuid'] if subworksheet_info else None
-    value = tokens_to_string(value_obj) if type == TYPE_DIRECTIVE else value_obj
-    if not value: value = ''  # TODO: change tables.py so that None's are allowed
-    return (bundle_uuid, subworksheet_uuid, value, type)
+    (bundle_info, subworksheet_info, value_obj, item_type) = item
+    return (
+        bundle_info['uuid'] if bundle_info else None,
+        subworksheet_info['uuid'] if subworksheet_info else None,
+        (tokens_to_string(value_obj) if item_type == TYPE_DIRECTIVE else value_obj) or '',
```
Comment (percyliang) [inline codalab/lib/worksheet_util.py:120]: Keep TODO

File: codalab/lib/worksheet_util.py [hunk 2]
```diff
@@ -249,49 +249,62 @@ def get_worksheet_uuid(client, base_worksheet_uuid, worksheet_spec):
...
-            return directive_item(string_to_tokens(m.group(1)))
+    line_types = [get_line_type(line) for line in form_result]
 
-        return markup_item(line)
+    # Extract bundle specs and resolve uuids in one batch
+    # bundle_specs = (line_indices, bundle_specs)
+    bundle_specs = zip(*[(i, BUNDLE_REGEX.match(line).group(3))
+                    for i, line in enumerate(form_result)
+                    if line_types[i] == TYPE_BUNDLE])
+    bundle_uuids = dict(zip(bundle_specs[0], get_bundle_uuids(client, worksheet_uuid, bundle_specs[1])))
```
Comment (percyliang) [inline codalab/lib/worksheet_util.py:283]: Comment what's in bundle_uuids (because it's not just a list of uuids!)

**Comment by percyliang** [comment]
LGTM!

**Comment by kashizui** [comment]
Fixes #185 
also Fixes #198

**Pull request merged**